src/server/hook_func.c

```diff
@@ -6012,128 +6063,124 @@ add_pending_mom_allhooks_action(void *minfo, unsigned int action)
 {
 	hook			*phook;
 
-
-	phook = (hook *)GET_NEXT(svr_execjob_begin_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_begin_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_prologue_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_prologue_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_epilogue_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_epilogue_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_end_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_end_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_preterm_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_preterm_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_exechost_periodic_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_exechost_periodic_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_exechost_startup_hooks);
-	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_exechost_startup_hooks);
-	}
-
-	phook = (hook *)GET_NEXT(svr_execjob_launch_hooks);
+	phook = (hook *)GET_NEXT(svr_allhooks);
 	while (phook) {
-		add_pending_mom_hook_action((mominfo_t *)minfo,
-			phook->hook_name, action);
-		phook = (hook *)GET_NEXT(phook->hi_execjob_launch_hooks);
+		if (phook->hook_name &&	(phook->event & MOM_EVENTS)) {
+			add_pending_mom_hook_action((mominfo_t *)minfo,
```

Comment:
merge to one line - these days we are leaning towards longer line lengths

Reply:
Done https://github.com/openpbs/openpbs/pull/1992/commits/d54ce17afb4e9d6cd1a3da070504726722e60e5c